solutionbox/code_free_ml/mltoolbox/code_free_ml/data/trainer/task.py

```diff
@@ -0,0 +1,849 @@
+  parser.add_argument('--analysis-output-dir',
+                      type=str,
+                      required=True,
+                      help=('Output folder of analysis. Should contain the'
+                            ' schema, stats, and vocab files.'
+                            ' Path must be on GCS if running'
+                            ' cloud training.'))
+  parser.add_argument('--run-transforms',
+                      action='store_true',
+                      default=False,
+                      help=('If used, input data is raw csv that needs '
+                            'transformation.'))
+
+  # HP parameters
+  parser.add_argument('--learning-rate', type=float, default=0.01,
+                      help='tf.train.AdamOptimizer learning rate')
+  parser.add_argument('--epsilon', type=float, default=0.0005,
+                      help='tf.train.AdamOptimizer epsilon')
+
+  # Model problems
+  parser.add_argument('--model-type',
+                      choices=['linear_classification', 'linear_regression',
+                               'dnn_classification', 'dnn_regression'],
+                      required=True)
+  parser.add_argument('--top-n',
+                      type=int,
+                      default=1,
+                      help=('For classification problems, the output graph '
+                            'will contain the labels and scores for the top '
+                            'n classes.'))
+  # Training input parameters
+  parser.add_argument('--max-steps', type=int, default=5000,
+                      help='Maximum number of training steps to perform.')
+  parser.add_argument('--num-epochs',
+                      type=int,
+                      help=('Maximum number of training data epochs on which '
+                            'to train. If both --max-steps and --num-epochs '
+                            'are specified, the training job will run for '
+                            '--max-steps or --num-epochs, whichever occurs '
+                            'first. If unspecified will run for --max-steps.'))
+  parser.add_argument('--train-batch-size', type=int, default=1000)
+  parser.add_argument('--eval-batch-size', type=int, default=1000)
+  parser.add_argument('--min-eval-frequency', type=int, default=100,
+                      help=('Minimum number of training steps between '
+                            'evaluations'))
+
+  # other parameters
+  parser.add_argument('--save-checkpoints-secs', type=int, default=600,
+                      help=('How often the model should be checkpointed/saved '
+                            'in seconds'))
+
+  args, remaining_args = parser.parse_known_args(args=argv[1:])
+
+  # All HP parambeters must be unique, so we need to support an unknown number
+  # of --layer_size1=10 --layer_size2=10 ...
+  # Look at remaining_args for layer_size\d+ to get the layer info.
+
+  # Get number of layers
+  pattern = re.compile('layer-size(\d+)')
```

Comment:
Layers that are not input or output layers, are hidden layers. See https://en.wikipedia.org/wiki/Feedforward_neural_network.